docs/architecture/dapr-for-net-developers/publish-subscribe-buildingblock.md

```diff
@@ -0,0 +1,398 @@
+
+## Reference architecture: eShopOnDapr
+
+As mentioned earlier in an earlier chapter, the accompanying [eShopOnDapr](https://github.com/dotnet-architecture/eShopOnDapr) app provides an end-to-end reference architecture for constructing a microservices application implementing Dapr. eShopOnDapr is an evolution of the widely popular [eShopOnContainers](https://github.com/dotnet-architecture/eShopOnContainer) app, created several years ago. Both versions use the Publish/Subscribe pattern for communicating [integration events](https://devblogs.microsoft.com/cesardelatorre/domain-events-vs-integration-events-in-domain-driven-design-and-microservices-architectures/#integration-events) across microservices. Integration events include:
+
+- When a user checks-out a shopping basket.
+- When a payment for an order has succeeded.
+- When the grace-period of a purchase has expired.
+
+Eventing in eShopOnContainers is based on the following `IEventBus` interface:
+
+```csharp
+public interface IEventBus
+{
+    void Publish(IntegrationEvent @event);
+
+    void Subscribe<T, TH>()
+        where T : IntegrationEvent
+        where TH : IIntegrationEventHandler<T>;
+}
+```
+
+Concrete implementations of this interface exist in eShopOnContainers for both RabbitMQ and Azure Service Bus. Each implementation included a great deal of custom plumbing code that was complex to understand and difficult to maintain.
+
+The newer eShopOnDapr significantly simplifies Publish/Subscribe behavior by using Dapr. For example, the `IEventBus` interface was reduced to a single method:
+
+```csharp
+public interface IEventBus
+{
+    Task PublishAsync(IntegrationEvent @event);
+}
+```
+
+### Publishing events
+
+In the updated eShopOnDapr, a single `DaprEventBus` implementation can support any Dapr-supported message broker. The following code block shows the simplified Publish method. Note how the `PublishAsync` method uses the Dapr client to publish an event:
+
+```csharp
+public class DaprEventBus : IEventBus
+{
+    private const string DAPR_PUBSUB_NAME = "pubsub";
+
+    private readonly DaprClient _dapr;
+    private readonly ILogger<DaprEventBus> _logger;
+
+    public DaprEventBus(DaprClient dapr, ILogger<DaprEventBus> logger)
+    {
+        _dapr = dapr;
+        _logger = logger;
+    }
+
+    public async Task PublishAsync<TIntegrationEvent>(TIntegrationEvent @event)
+        where TIntegrationEvent : IntegrationEvent
+    {
+        var topicName = @event.GetType().Name;
+
+        _logger.LogInformation("Publishing event {@Event} to {PubsubName}.{TopicName}", @event, DAPR_PUBSUB_NAME, topicName);
+
+        // Make sure to pass the concrete event type to PublishEventAsync,
+        // which can be accomplished by casting the event to dynamic. This ensures
+        // that all event fields are properly serialized.
+        await _dapr.PublishEventAsync(DAPR_PUBSUB_NAME, topicName, (dynamic)@event);
+    }
```

Comment:
I agree about the name. I will update that in the code and the snippet.
String interpolation is not used in log statements because eShopOnDapr uses Serilog as logging framework. And by specifying the variables like this, they will be exposed as separate (searchable) properties on each log-record (see https://github.com/serilog/serilog/wiki/Writing-Log-Events).